### ISSUE TITLE
RTX 3090 very slow face extraction...??

### ISSUE THREAD

**Issue by testerson34** [iperov/DeepFaceLab#5243]
## Expected behavior

I am using `data_src faceset extract` and `data_dst faceset extract` faces from frames.  I expect it to perform as good or better than my RTX 2060 SUPER.

## Actual behavior

Extraction is extremely slow when compared to the RTX 2060 SUPER in my other machine.  I've been using it extract faces and then move to my 3090 rig.  

## Steps to reproduce

Simply run the face extraction tools and notice that the faces are extracted much more slowly.

**Comment by Obsyden** [comment]
Same issue here with a Linux Mint 20.1 setup - possibly to do with new drivers

**Comment by Eleganze** [comment]
Same her. Down to 1 it/s on 3090. Regular 1080ti much faster. Something with drivers or Tensorflow?

**Comment by holycowdude** [comment]
Extremely slow face extraction using GPU on my RTX3090 as well.
Often i get an error "GPU does not respond" and it quits, when it does work its much slower than by CPU

**Comment by spullara** [comment]
Mine claims it is using the GPU (RTX 2080 Ti) but it is actually using the CPU based on task manager. My guess is that it is falling back to CPU.

**Comment by Quieky** [comment]
weird... face extraction is the only thing that works pretty properly on my rtx 3090... faster than my "old" gtx 1080 ti... just everything else doesnt (yet)... (running my rtx 3090 with the new amd ryzen 9 5950x... not sure, if thats important here somehow)

**Comment by testerson34** [comment]
@Quieky That's the exact setup I have; Ryzen 9 5950x + RTX 3090.  Running DFL 01_04_2020.  You're the only person I've seen say the extraction is working well on the 3090 regardless of the CPU.

**Comment by zabique** [comment]
![Screenshot_20210204-091530_TeamViewer](https://user-images.githubusercontent.com/58276510/106871352-0c41d500-66ca-11eb-9c10-64b1c2e1d881.jpg)

No issues here

**Comment by testerson34** [comment]
**TL;DR** - It seems that the 3090 is slower to start, and it runs in "chunks", but overall it actually is faster than my 2060 SUPER.

So this is odd.  It seems to be a mirage now that I'm looking at this more closely.  I have two machines I'm working with here.

* Intel i9 10900K w/ RTX 2060 SUPER
* AMD Ryzen 9 5950X w/ RTX 3090

I'm extracting from a clip with 23,840 frames.  I prepared the same clip on both machines, and I hit ENTER at the exact time on each to start the extraction process.

The 2060 gets started much more quickly.  It was about 200 frames in before the 3090 even started running.

Also, the 2060 goes at a nice, regular flow.  You can watch the frame counter move along, and while sometimes it will go a little bit more quickly and then slow down again, it never stops completely.

The 3090 will run for a little bit, then stop completely, then run for a bit more, stop, etc.  When it runs, though, it will run through quite a few frames much more quickly.

Over time, though, the 3090 has caught up to the 2060, and has surpassed it.  My 3090 as of this writing is at 7500 frames while the 2060 is at 7200.

The it/s jump around quite a bit on both, but they seem to have settled at about 2.51it/s on the 3090 while the 2060 is showing 2.39it/s.

Just wanted to provide the additional feedback.  I guess I'm going to go ahead and stick with 3090 for extraction after all.  I had been using the 2060 for that and then transferring for training, but that seems to actually be slower overall.

**Comment by testerson34** [comment]
Info I'm getting elsewhere says that with CUDA v11.1 it introduces the stuff necessary to take full advantage of RTX 30 series.  I'm currently running v11.2 on the system, but I can't tell if that's what DFL is seeing or not.  Does it use the system installed version, or does it use its own version included with it?

**Comment by zabique** [comment]
DFL use dll in environment, I did swap existing ones with latest cuda_11.2.0 and cudnn-11.2-windows-x64-v8.1.0.77, still no difference at all for ampere.

**Comment by ayakashi-ml** [comment]
> Info I'm getting elsewhere says that with CUDA v11.1 it introduces the stuff necessary to take full advantage of RTX 30 series. I'm currently running v11.2 on the system, but I can't tell if that's what DFL is seeing or not. Does it use the system installed version, or does it use its own version included with it?

You cannot use CUDA 11.2 if your Tensorflow is not compiled with 11.2 support. The current version of Tensorflow only supports 11.0, I believe.

In order to use CUDA 11.2, you must compile Tensorflow with CUDA 11.2 and CuDNN 8.1. I encountered a similar problem with Faceswap, and I simply swapped out the Tensorflow that came normally shipped with it with a wheel compiled for CUDA 11.1. 

I have yet to try the same thing with DFL. Just a pointer.
